org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/SchemaLocation.java

```diff
@@ -12,40 +12,62 @@
  */
 package org.eclipse.lemminx.dom;
 
+import java.util.Collection;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.StringTokenizer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
- * 
+ *
  * The declared "xsi:schemaLocation"
  */
 public class SchemaLocation {
 
-	private final Map<String, String> schemaLocationValuePairs;
+	private final Map<String, SchemaLocationHint> schemaLocationValuePairs;
 
 	private final DOMAttr attr;
 
+	// The text to match is of the form:
+	// http://example.org root.xsd http://example.org bison.xsd http://example.org potato.xsd
+	private static final Pattern SCHEMA_LOCATION_PAIR_PATTERN = Pattern.compile("\\s*([^\\s]+)\\s+([^\\s]+)\\s*");
+
 	public SchemaLocation(String base, DOMAttr attr) {
 		this.attr = attr;
 		this.schemaLocationValuePairs = new HashMap<>();
 		String value = attr.getValue();
-		StringTokenizer st = new StringTokenizer(value);
-		do {
-			String namespaceURI = st.hasMoreTokens() ? st.nextToken() : null;
-			String locationHint = st.hasMoreTokens() ? st.nextToken() : null;
+		Matcher locPairMatcher = SCHEMA_LOCATION_PAIR_PATTERN.matcher(value);
+		while (locPairMatcher.find()) {
+			String namespaceURI = locPairMatcher.group(1);
+			String locationHint = locPairMatcher.group(2);
 			if (namespaceURI == null || locationHint == null)
 				break;
-			schemaLocationValuePairs.put(namespaceURI, locationHint);
-		} while (true);
+			DOMNode valNode = attr.getNodeAttrValue();
+			// matcher matches 1 char ahead, but valNode's start is at beginning of "
+			int start = valNode.getStart() + locPairMatcher.start(2);
+			// matcher matches end correctly, but range needs to be one past end
+			// to highlight the end, and one to make up for "
+			int end = valNode.getStart() + locPairMatcher.end(2) + 2;
+			schemaLocationValuePairs.put(namespaceURI,
+					new SchemaLocationHint(start, end, locationHint, this));
+		}
 	}
 
 	public String getLocationHint(String namespaceURI) {
-		return schemaLocationValuePairs.get(namespaceURI);
+		return schemaLocationValuePairs.get(namespaceURI).getHint();

```

Comment:
You must prevent from NPE because namespaceURI could not be defined in the xsi:schemaLocation. Do liek this:

```java
public String getLocationHint(String namespaceURI) {
	SchemaLocationHint schemaLocationHint = schemaLocationValuePairs.get(namespaceURI); 
	return schemaLocationHint != null ? schemaLocationHint.getHint() : null;
}
```

Reply:
Thank you!